MOBILE_CLIENT/COMMON/_HIGH/OVERLAY_VULNERABILITY/recommendation.md

```diff
@@ -0,0 +1,4 @@
+
+# Overlay Vulnerability
+
+To mitigate overlay vulnerabilities in mobile applications, developers should implement robust security measures with a focus on enhancing settings and user interactions. First and foremost, ensure that user inputs and application settings undergo thorough validation and filtering to prevent potential exploitation by malicious overlays. Pay special attention to sensitive settings that can be manipulated by overlays, such as authentication parameters or access controls. Implement strict input validation checks on user-modifiable settings to prevent unauthorized changes. When handling settings related to user authentication or authorization, enforce strong authentication mechanisms and authorization controls to minimize the impact of potential overlay attacks. Provide clear user guidance within the application on recognizing and avoiding suspicious overlays. Regularly update the application to address any newly identified vulnerabilities, and consider implementing overlay detection mechanisms to enhance the application's resilience against potential threats.
```

Comment:
Separate points would be better than one giant paragraph

Reply:
You can mention this:

Newer versions of Android have introduced several mitigations, including the following:

- On Android 6 (API level 23) and higher, users have to explicitly grant the permission for the app to create an overlay window.
- On Android 12 (API level 31) and higher, apps can pass true into [Window.setHideOverlayWindows()](https://developer.android.com/reference/android/view/Window#setHideOverlayWindows(boolean)).

Reply:
noted